paper_framework_sources/p2203_bulk_schedule.md

```diff
@@ -354,14 +352,21 @@ based on prior work, each of these user-visible algorithms would take a policy.
 They are all user-facing as well as implementation-details.
 In the `get_execution_policy` model each of these can communicate a policy to the prior algorithm.
 
-These algorithms are communicating single values rather than bulk.
-If we want to be sure that we can implement a completion signal safely, that is useful because it allows the executor to guarantee safe communication with the next executor.
+We do realise that execution policy is a broader concept than just forward progress - it may be that we need to separate concerns here.
+The aspect of the execution policy that is a minimum requirement for the code being passed in - the way they are used in the current set of defined policies on the parallel algorithms - may be abstracted in this way while the wider policy is separate and describing execution may be useful.
+In that case `get_forward_progress_requirement` might be a better name.
```

Comment:
I think there is possibly also another aspect bound up in the constraints/policy in addition to forward-progress, and that is "thread safety". e.g. if a receiver returns a policy of `par`, then it's not only saying that it requires concurrent forward progress guarantee for threads that call it, but also that it is safe to call concurrently from multiple threads.